PocketKit/Sources/SharedPocketKit/User.swift

```diff
@@ -8,26 +8,44 @@ public enum Status: String {
 }
 
 public protocol User {
-    var status: Status? { get }
+    var status: Status { get }
+    var statusPublisher: Published<Status>.Publisher { get }
+    var publishedStatus: Published<Status> { get }
```

Comment:
what is the different between `statusPublisher` and `publishedStatus` and how are we expecting to use them?

Reply:
This relates to what we were discussing in the Premium Subscription architecture meeting: if we want to pass `User` (which is a protocol) to our view models etc, and, at the same time observe changes in `status`, then, since we cannot declare a property in a protocol with a property wrapper (this makes sense, since protocols are generic types, while property wrappers are concrete), then we need a publisher and a published to link, respectively, the projected and the wrapped value of that property, if we declare it as `@Published` in a concrete implementation of `User`. You can read more on the internet, for example [here](https://danielbernal.co/2020/06/26/combine-and-protocols-in-swift/) or [here](https://medium.com/swlh/how-to-define-a-protocol-with-published-property-wrapper-type-1b6349097064). That said, I'm not a big fan of this implementation and if we find a better way to achieve this result, I'd be up for it!

Reply:
Can we `sink` on `statusPublisher`, do we also need to expose `publishedStatus`? I'll probably have a better understanding when I see this in used. thanks for the explanation though!